matrix/matrix_operation.py

```diff
@@ -148,11 +163,13 @@ def _check_not_integer(matrix: list[list]) -> bool:
     raise TypeError("Expected a matrix, got int/list instead")
 
 
-def _shape(matrix: list[list]) -> list:
+def _shape(matrix: list[list]) -> tuple[int, int]:
     return len(matrix), len(matrix[0])
```

Comment:
In case of matrix = [[1, 2], [1, 2, 3]] it would return [2, 2]

Wouldn't this help?
`def _shape(matrix: list[list]) -> tuple[int, int]:`
`if (len(set([len(row) for row in matrix]) == 1):`
`return len(matrix), len(matrix[0])`
`raise TypeError("Matrix is not rectangular")`